src/test/java/core/basesyntax/service/impl/DataProcessingServiceImplTest.java

```diff
@@ -0,0 +1,40 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitShopOperation;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.interfaces.DataProcessingService;
+import core.basesyntax.service.interfaces.TransactionStrategy;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class DataProcessingServiceImplTest {
+    private TransactionStrategy transactionStrategy;
+    private DataProcessingService dataProcessingService;
+
+    @BeforeEach
+    void setUp() {
+        transactionStrategy = new TransactionStrategyImpl();
+        dataProcessingService = new DataProcessingServiceImpl(transactionStrategy);
+    }
+
+    @Test
+    void processData_OK() {
+        List<FruitTransaction> transactions = List.of(
+                new FruitTransaction(FruitShopOperation.BALANCE, "apple", 40),
```

Comment:
![image](https://github.com/mate-academy/jv-fruit-shop-tests/assets/82539553/d431b566-cc02-4d9f-8e47-7df1f0180a7d)

You can test only 1 operation here to verify if you call handler correctly, you don't need to test logic of strategy or handlers here.